webapp/channels/src/components/admin_console/team_channel_settings/team/details/index.ts

```diff
@@ -15,16 +16,25 @@ import {
     unlinkGroupSyncable,
     patchGroupSyncable,
 } from 'mattermost-redux/actions/groups';
+import {ActionFunc, GenericAction} from 'mattermost-redux/types/actions';
 
 import {setNavigationBlocked} from 'actions/admin_actions';
 
-import TeamDetails from './team_details';
+import TeamDetails, {Props as TeamDetailsProps} from './team_details';
+
+type OwnProps = {
+    match: {
+        params: {
+            teamID: string;
+        };
+    };
+}
```

Comment:
You can make use of types from react-router like this

https://github.com/mattermost/mattermost/blob/c943ed6859b1be013e4da92b1952c607a35258bc/webapp/channels/src/components/channel_layout/center_channel/index.ts#L26-L30